File: app/lib/ageGradient.ts
// Shared strike-age gradient: t=0 → oldest (dark purple), t=1 → newest (yellow).
// Used by the live map's dot layer and the biggest-storm replay.

type Stop = [pos: number, r: number, g: number, b: number, a: number];

const STOPS: Stop[] = [
  [0,    30,   0,   80, 0.20],
  [0.30, 120,  0,  160, 0.42],
  [0.55, 210,  10,  10, 0.65],
  [0.78, 255, 120,   0, 0.80],
  [1,    255, 230,   0, 0.92],
];

export function ageColor(t: number): [number, number, number, number] {
  let i = 0;
  while (i < STOPS.length - 2 && STOPS[i + 1][0] <= t) i++;
  const s0 = STOPS[i], s1 = STOPS[i + 1];
  const f = s1[0] > s0[0] ? (t - s0[0]) / (s1[0] - s0[0]) : 0;
  return [
    Math.round(s0[1] + f * (s1[1] - s0[1])),
    Math.round(s0[2] + f * (s1[2] - s0[2])),
    Math.round(s0[3] + f * (s1[3] - s0[3])),
    +(s0[4] + f * (s1[4] - s0[4])).toFixed(2),
  ];
}

File: lightning-stats/lib/utils/age_gradient.dart
import 'package:flutter/material.dart';

/// Shared strike-age gradient: t=0 -> oldest (dark purple), t=1 -> newest (yellow).
/// Ported 1:1 from app/lib/ageGradient.ts so live-map and replay dots match the web app.
class _Stop {
  final double pos;
  final int r, g, b;
  final double a;
  const _Stop(this.pos, this.r, this.g, this.b, this.a);
}

const List<_Stop> _stops = [
  _Stop(0, 30, 0, 80, 0.20),
  _Stop(0.30, 120, 0, 160, 0.42),
  _Stop(0.55, 210, 10, 10, 0.65),
  _Stop(0.78, 255, 120, 0, 0.80),
  _Stop(1, 255, 230, 0, 0.92),
];

Color ageColor(double t) {
  var i = 0;
  while (i < _stops.length - 2 && _stops[i + 1].pos <= t) {
    i++;
  }
  final s0 = _stops[i];
  final s1 = _stops[i + 1];
  final f = s1.pos > s0.pos ? (t - s0.pos) / (s1.pos - s0.pos) : 0.0;
  final r = (s0.r + f * (s1.r - s0.r)).round();
  final g = (s0.g + f * (s1.g - s0.g)).round();
  final b = (s0.b + f * (s1.b - s0.b)).round();
  final a = s0.a + f * (s1.a - s0.a);
  return Color.fromRGBO(r, g, b, a);
}
